chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -521,7 +521,13 @@ impl NetworkState {
                 })
             })
             .await;
-            this.routing_table_view.add_local_edges(&edges);
+            let new_local_edges = this.routing_table_view.add_local_edges(&edges);
+
+            // Local edge information is also kept in the Connection fields in peers.
+            // Make sure that it gets updated too.
+            for new_edge in new_local_edges {
+                this.tier2.update_edge(&new_edge);
+            }
```

Comment:
local_edges are the edges of the graph adjacent to this node.
edge stored in connection is the edge that the ends of the connection have agreed upon during handshake.

Given that now the edge is supposed to change dynamically and that it is delivered to a node via SyncRoutingTable message it doesn't make sense to duplicate it I guess. We can either:
- remove edge field from the connection entirely and just rely on local_edges (I think that would be preferred)
- revert to using ResponseUpdateNonce message, so that each connection can manage the connection on their own.

Reply:
yeah, we could remove the edge from the connection -- but I'd do that in a separate PR (as this one is becoming too large)

Reply:
ack, please add a TODO

Reply:
also you unnecesarily take a lock on the whole pool in Pool::update_edge. Instead just update the edge in-place.

Reply:
and for that you might want ArcMutex/ArcSwap instead of AtomicCell.